bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerBaseHandler.java

```diff
@@ -69,5 +69,31 @@ public void updateChannels() {
         }
     }
 
+    @SuppressWarnings("null")
+    protected void handleNextDurationCommand(ChannelUID channelUID, Command command) {
+        if (!(command instanceof QuantityType<?>)) {
+            logger.info("Ignoring implausible non-QuantityType command for NEXT_DURATION");
+            return;
+        }
+        QuantityType<?> quantity = (QuantityType<?>) command;
+        nextDurationTime = quantity.toUnit(Units.SECOND).toBigDecimal();
```

Comment:
`toUnit` can return null if the unit can't be converted.

Reply:
DONE